### ISSUE TITLE
Improve screen reader accessibility of default themes

### ISSUE THREAD

**Issue by sukiletxe** [getpelican/pelican#1907]
Hello,
Currently, most of the themes (if not all of them) in [pelican-themes](http://github.com/getpelican/pelican-themes) suffer from accessibility problems for screen reader users. Don't get me wrong, I don't mean they are unusable for us. I mean that they are more difficult to use, because they lack or missuse some features introduced in HTML5 (`role`attributes or their equivalent `<nav>`, `<main>`, `<aside>`), and structure the document's headings in non-structured ways (e. g. using css instead of headings and vice-versa).
The default themes also have this first problem. Considering that the themes are sometimes built based on the default ones, I think it would be benefficial to solve this.
If you are interested, I will create a pr with the necessary changes.
In addition, if any contributor to [pelican-themes](http://github.com/getpelican/pelican-themes) is interested in this, he / she can contact me.

**Comment by mitchtbaum** [comment]
> The default themes also have this first problem.  Considering that the themes are sometimes built based on the default ones, I think it would be beneficial to solve this.

I agree with this approach very much, getting the most needed gains in one swift act, ie by implementing best practices that work closer toward maximum usability.  I hope to solve this through #1280 instead of isolated fixes to one hard-coded theme or another.  A review of [mitchtbaum/base](https://github.com/mitchtbaum/base) and, to some degree, [mitchtbaum/fixed_sidebar](https://github.com/mitchtbaum/fixed_sidebar), would help make sure this work can hit its targets using its current layouts' semantics and libraries I picked, each pretty well-researched for a staring point.  By a quick glance, it looks like I implemented ```<nav>``` and ```<aside>``` well, and I could add ```<main>``` to [mitchtbaum/base/templates/base.html#L14](https://github.com/mitchtbaum/base/blob/master/templates/base.html#L14).  Any additional improvements to this new theme system's HTML, CSS, Jinja chunks (currently called widgets), underlying Python code, or new layouts (themes) would help as well.

**Comment by sukiletxe** [comment]
Hello,
Browsing through Base, I can see it's highly modular, which I don't 
think it's not the standard in themes in Pelican. Nontheless, I can see 
its advantages.
However, I can also see that some widgets are repeated in parts of 
others (like categories and some parts of articles), and that English 
and French appear mixed.
By the way, the cattegories list, in my opinion should be an <aside>. 
However, opinions may vary (see Wikipedia,for example, which never uses it).
The pagination widget could be improved by changing the symbols with 
actual text, or hiding the symbols to screen readers and showing 
invissible text instead. Similarly, skip links before the header could 
be useful (which only went vissible when focused with the tab key), and 
an invissible "post navigation" heading inside the <nav>.
If you wish, I can send a pr with those changes.

**Comment by mitchtbaum** [comment]
Thanks for this initial review.  I started making a few of these changes and some others in getting this ready for development again.  I renamed it and moved it to [Salutations Theme](https://github.com/getpelican-themes/salutations), so let's move over to that project.

> some widgets are repeated in parts of others (like categories and some parts of articles)

I mostly copied everything from [Lightweight Theme](https://github.com/getpelican-themes/lightweight) and so far focused on adapting it to this style of component-orientation.  I missed that error, so I would appreciate your help to fix it, and please also point out anything you believe would make this theme more functional and would improve its usability.  That will hopefully ripple out for many more themes.

> the categories list, in my opinion, should be an `<aside>`

Please create an issue for this one first, because I would like to understand it a little better.  From searching, "side-navigation" aside, this seems to have some interesting significance.

> The pagination widget could be improved by changing the symbols with actual text, or hiding the symbols to screen readers and showing invisible text instead.

Please tell me about which you would prefer in https://github.com/getpelican-themes/salutations/issues/8

> Similarly, skip links before the header could be useful (which only went visible when focused with the tab key), and an invisible "post navigation" heading inside the `<nav>`.

Sounds good.  Please submit PRs for these too.

**Comment by sukiletxe** [comment]
Okay, let's move on to that project then. As for the categories list 
issue, as it is over the main content area (which I would considerthe 
part where the other files start appending to `block content`), I think 
it's Okay if it is left as is. My only base for this is that blogger and 
wordpress only apply `<aside>` to widgets below the main content, and 
`<nav>` otherwhise.

**Comment by justinmayer** [comment]
Closing due to inactivity. Agree that we should do what we can to improve accessibility. Please feel free to comment here if you feel the issue should be re-opened.
